pkg/download/go_lister.go

```diff
@@ -0,0 +1,75 @@
+package download
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/module"
+	"github.com/gomods/athens/pkg/storage"
+	"github.com/spf13/afero"
+)
+
+type Lister func(string) (*storage.RevInfo, []string, error)
+
+type listResp struct {
+	Path     string
+	Version  string
+	Versions []string `json:",omitempty"`
+	Time     time.Time
+}
+
+// NewGoLister bla
```

Comment:
@marpio would you mind filling out the godocs here? we can save this for a follow-up if you want too